### PR TITLE
Make Search Models More Configurable. Upgrade Default Cross-Encoder

### PR DESCRIPTION
**Pull request by debanjum**
- 9198836b Upgrade default cross-encoder to mixedbread ai's mxbai-rerank-xsmall
- d0456cd0 Support more embedding models by making query, docs encoding configurable

### CODE REVIEW

File: src/khoj/processor/embeddings.py [hunk 1]
```diff
@@ -25,9 +25,15 @@ def __init__(
...
+        docs_encode_kwargs: dict = {},
+        model_kwargs: dict = {},
     ):
-        self.encode_kwargs = {"normalize_embeddings": True}
-        self.model_kwargs = {"device": get_device()}
+        default_query_encode_kwargs = {"show_progress_bar": False, "normalize_embeddings": True}
+        default_docs_encode_kwargs = {"show_progress_bar": True, "normalize_embeddings": True}
+        self.query_encode_kwargs = merge_dicts(query_encode_kwargs, default_query_encode_kwargs)
+        self.docs_encode_kwargs = merge_dicts(docs_encode_kwargs, default_docs_encode_kwargs)
+        self.model_kwargs = merge_dicts(model_kwargs, {"device": get_device()})
```
Comment (sabaimran) [inline src/khoj/processor/embeddings.py:36]: It seems like this isn't being used?

File: documentation/docs/miscellaneous/advanced.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ sidebar_position: 3
...
 For example, the [paraphrase-multilingual-MiniLM-L12-v2](https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2) supports [50+ languages](https://www.sbert.net/docs/pretrained_models.html#:~:text=we%20used%20the%20following%2050%2B%20languages), has good search quality and speed. To use it:
-1. Manually update the search config in server's admin settings page. Go to [the search config](http://localhost:42110/server/admin/database/searchmodelconfig/). Either create a new one, if none exists, or update the existing one. Set the bi_encoder to `sentence-transformers/multi-qa-MiniLM-L6-cos-v1` and the cross_encoder to `cross-encoder/ms-marco-MiniLM-L-6-v2`.
+1. Manually update the search config in server's admin settings page. Go to [the search config](http://localhost:42110/server/admin/database/searchmodelconfig/). Either create a new one, if none exists, or update the existing one. Set the bi_encoder to `sentence-transformers/multi-qa-MiniLM-L6-cos-v1` and the cross_encoder to `mixedbread-ai/mxbai-rerank-xsmall-v1`.
```
Comment (sabaimran) [inline documentation/docs/miscellaneous/advanced.md:10]: Should we add some guidance for using the new specialized args?

File: src/khoj/database/models/__init__.py [hunk 3]
```diff
@@ -182,7 +182,10 @@ class ModelType(models.TextChoices):
     name = models.CharField(max_length=200, default="default")
     model_type = models.CharField(max_length=200, choices=ModelType.choices, default=ModelType.TEXT)
     bi_encoder = models.CharField(max_length=200, default="thenlper/gte-small")
-    cross_encoder = models.CharField(max_length=200, default="cross-encoder/ms-marco-MiniLM-L-6-v2")
+    bi_encoder_model_config = models.JSONField(default=dict)
+    bi_encoder_query_encode_config = models.JSONField(default=dict)
+    bi_encoder_docs_encode_config = models.JSONField(default=dict)
```
Comment (sabaimran) [inline src/khoj/database/models/__init__.py:187]: Should we add some comments/a link explaining what these would be used for?

File: src/khoj/processor/embeddings.py (hunk 1 above)
Comment (debanjum) [reply src/khoj/processor/embeddings.py:36]: It is being used when we initialize the sentence-transformer model

File: src/khoj/database/models/__init__.py (hunk 3 above)
Comment (debanjum) [reply src/khoj/database/models/__init__.py]: Added

**Pull request merged**